tests/testthat/test-generate_data.R

```diff
@@ -705,3 +693,113 @@ test_that("genSynthetic works.", {
+
+  expect_error(genSurv(dtSurv, sdef, digits = "a"), class = "simstudy::wrongClass")
+  expect_error(genSurv(dtSurv, sdef, digits = c(1,2,3)), class = "simstudy::lengthMismatch")
+
+  expect_error(genSurv(dtSurv, sdef, idName = "ID"), class = "simstudy::notDefined")
+})
+
+test_that("genSpline doesn't throw any errors", {
+  
+  skip_on_cran()
+  
+  ddef <- defData(varname = "x1", formula = "0;1", dist = "uniform")
+  theta1 <- c(0.1, 0.8, 0.6, 0.4, 0.6, 0.9, 0.9)
+
+  knots <- c(0.25, 0.5, 0.75)
+  expect_silent(viewSplines(knots = knots, theta = theta1, degree = 3))
+
+  dt <- genData(1000, ddef)
+  expect_silent(genSpline( dt = dt, newvar = "weight",
+                 predictor = "x1", theta = theta1,
+                 knots = knots, degree = 3,
+                 noise.var = .025))
+  
+  ddef <- defData(varname = "x1", formula = "20;60", dist = "uniform")
+  dt <- genData(1000, ddef)
+  expect_silent(genSpline( dt = dt, newvar = "weight",
+                           predictor = "x1", theta = theta1,
+                           knots = knots, degree = 3,
+                           noise.var = .025))
+  
+  expect_silent(genSpline( dt = dt, newvar = "weight",
+                           predictor = "x1", theta = theta1,
+                           knots = knots, degree = 3,newrange = "0;10",
+                           noise.var = .025))
+  
+})
+
+test_that("genSpline throws errors", {
+  skip_on_cran()
+  
+  ddef <- defData(varname = "x1", formula = "0;1", dist = "uniform")
+  theta1 <- c(0.1, 0.8, 0.6, 0.4, 0.6, 0.9, 0.9)
+  
+  knots <- c(0.25, 0.5, 0.75)
+  
+  dt <- genData(1000, ddef)
+  expect_error(genSpline( dt = ddd, newvar = "weight",
+                           predictor = "x1", theta = theta1,
+                           knots = knots, degree = 3,
+                           noise.var = .025), regexp = "Data table does not exist.")
+  expect_error(genSpline( dt = dt, newvar = "weight",
+                          predictor = "x2", theta = theta1,
+                          knots = knots, degree = 3,
+                          noise.var = .025), regexp = "not in data.table")
+  expect_error(genSpline( dt = dt, newvar = 5,
+                          predictor = "x1", theta = theta1,
+                          knots = knots, degree = 3,
+                          noise.var = .025), regexp = "newvar must be a string")
+  expect_error(genSpline( dt = dt, newvar = "weight",
+                          predictor = "x1", theta = theta1,
+                          knots = knots, degree = 3, newrange = "4;3;2",
+                          noise.var = .025), regexp = "Range not specified as two values")
+  # expect_error(genSpline( dt = dt, newvar = "weight",
+  #                         predictor = "x1", theta = theta1,
+  #                         knots = knots, degree = 3, newrange = "1;a",
+  #                         noise.var = .025), regexp = "Non-numbers entered in range")
```

Comment:
If this test does not work I would remove it xor create a github issue to fix the underlying problem

Reply:
The test works but it also generates a warning. I will keep commented out, but will add to issues.